x-pack/plugins/apm/public/components/shared/Stacktrace/index.tsx

```diff
@@ -42,12 +43,22 @@ export function Stacktrace({ stackframes = [], codeLanguage }: Props) {
         if (group.isLibraryFrame) {
           const initialVisiblity = groups.length === 1; // if there is only a single group it should be visible initially
           return (
-            <LibraryStackFrames
-              key={i}
-              initialVisiblity={initialVisiblity}
-              stackframes={group.stackframes}
-              codeLanguage={codeLanguage}
-            />
+            <Fragment>
+              {group.stackframes.length > 1 && i !== 0 && (
+                <EuiSpacer size="m" />
+              ) // render a leading spacer if LibraryStackFrames is collapsible and it's not the first item
+              }
+              <LibraryStackFrames
+                key={i}
+                initialVisiblity={initialVisiblity}
+                stackframes={group.stackframes}
+                codeLanguage={codeLanguage}
+              />
+              {group.stackframes.length > 1 && i !== groups.length - 1 && (
+                <EuiSpacer size="m" />
+              ) // render a trailing spacer if LibraryStackFrames is collapsible and it's not the last item
+              }
+            </Fragment>
```

Comment:
What do you think about extracting the logic?

```tsx
// render a leading spacer if LibraryStackFrames is collapsible and it's not the first item
const hasLeadingSpacer = group.stackframes.length > 1 && i !== 0;

// render a trailing spacer if LibraryStackFrames is collapsible and it's not the last item
const hasTrailingSpacer = group.stackframes.length > 1 && i !== groups.length - 1;

return (
  <Fragment>
    {hasLeadingSpacer && <EuiSpacer size="m" />}
    <LibraryStackFrames
      key={i}
      initialVisiblity={initialVisiblity}
      stackframes={group.stackframes}
      codeLanguage={codeLanguage}
    />
    {hasTrailingSpacer && <EuiSpacer size="m" />}
  </Fragment>
);
```

You could even drop the comments like:

```js
const isCollapsible = group.stackframes.length > 1;
const hasLeadingSpacer = isCollapsible && i !== 0;
const hasTrailingSpacer = isCollapsible && i !== groups.length - 1;
```